src/main/java/tigase/mongodb/muc/MongoHistoryProvider.java

```diff
@@ -116,6 +128,57 @@ public void addSubjectChange(Room room, Element message, String subject, JID sen
 	                             Date time) {
 	}
 
+	@Override
+	public Item getItem(BareJID owner, String stableId) throws RepositoryException {
+		try {
+			byte[] rid = generateId(owner);
+			Document result = historyCollection.find(
+							Filters.and(Filters.eq("room_jid_id", rid), Filters.eq("stable_id", UUID.fromString(stableId))))
+					.projection(Projections.include("msg", "timestamp"))
+					.first();
+			if (result == null) {
```

Comment:
I'm wondering if `getItem()` methods shouldn't be either `Optional<>` or annotated with `@Nullable` to be more explicit that the item could indeed not exist.

Reply:
Maybe we should just throw `NoSuchElementException`? I'm not convinced that we should use optionals here. If so, we can consider that in 9.0 when we will deal with other `nullable` values being returned (ie. from `Element` class).

Reply:
OK, let's leave it for 9.0.

I wouldn't throw `NoSuchElementException`...